numpyro/contrib/autoguide/__init__.py

```diff
@@ -21,8 +21,9 @@
     UnpackTransform,
     biject_to
 )
-from numpyro.distributions.util import sum_rightmost
+from numpyro.distributions.util import cholesky_inverse, sum_rightmost
 from numpyro.infer_util import constrain_fn, find_valid_initial_params, init_to_median, transform_fn
+import numpyro.infer.svi
```

Comment:
The second time I faced a circular import issue in NumPyro: we call `elbo` which belongs to `svi.py` file, in that file, we call `AutoContinuous`, which is this file. >"<

Reply:
Yeah ideally the SVI class shouldn't need to know about AutoContinuous, but I don't have a better solution to that issue.

Reply:
My suggestion would be to do the following:
 - Refactor ELBO to make it a class.
 - Subclass ELBO in Autoguide to use `base_param_map`, we can call this method `Autoguide_ELBO`, or something.

That way we can inherit any other functionality from the base ELBO class later. That way, also all contrib code also remains in contrib and isn't imported by core NumPyro.

Reply:
Agreed! We should distinguish the name when the meaning of ELBO in the usual guide and the auto guide are different (one is in constrained space, the other is in unconstrained space). I remember that you already suggested this previously.